src/main/java/qna/domain/Answer.java

```diff
@@ -68,6 +70,20 @@ public void toQuestion(Question question) {
         this.question = question;
     }
 
+    private void validationOwner(User user) throws CannotDeleteException {
+        if (!this.isOwner(user)) {
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+        }
+    }
+
+    public DeleteHistories delete(User user, DeleteHistories deleteHistories) throws CannotDeleteException {
+        validationOwner(user);
+        setDeleted(true);
+
+        deleteHistories.addHistory(new DeleteHistory(ContentType.ANSWER, this.getId(), this.getWriter(), LocalDateTime.now()));
+        return deleteHistories;
+    }
```

Comment:
DeleteHistories를 파라미터로 받을 필요가 있을지 고민해보아요 :)
일급컬렉션인 Answers에서 DeleteHistories를 반환해주면 어떨까요 :) ?
```suggestion
    public DeleteHistory delete(User user) throws CannotDeleteException {
        validationOwner(user);
        setDeleted(true);
        return new DeleteHistory(ContentType.ANSWER, this.getId(), this.getWriter(), LocalDateTime.now());
    }
```

Reply:
Answers에서 이력을 추가하면되는 내용인데 굳이 파라미터를 받아서 내부에서 처리를 하고있었네요
DeleteHistory를 반환하고 외부에서 DeleteHistories에 넣도록 변경하였습니다.